entrypoints/air_pollution_shortest_path.py

```diff
@@ -1,13 +1,106 @@
 """Find the least cost path from source to target by minimising air pollution."""
 
-import routex as rx
-import urbanroute as ur
+from typing import Optional, Tuple
+import logging
+import typer
+import geopandas as gpd
+import osmnx as ox
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+import matplotlib.colors as colors
 
+import networkx as nx
+import math
+from cleanair.databases.queries import AirQualityResultQuery
+from cleanair.loggers import get_logger
+from shapely.geometry import Polygon
 
-def main():
-    """Main method for entrypoint."""
-    pass
+from routex.types import Node
+from urbanroute.geospatial import update_cost, ellipse_bounding_box
+from urbanroute.queries import HexGridQuery
+
+
+def main(  # pylint: disable=too-many-arguments
+    secretfile: str,
+    instance_id: str = "d5e691ef9a1f2e86743f614806319d93e30709fe179dfb27e7b99b9b967c8737",
+    source: Tuple[float, float] = (51.4929, -0.1215),
+    start_time: Optional[str] = "2020-01-24T09:00:00",
+    target: Tuple[float, float] = (51.4929, -0.2215),
+    upto_time: Optional[str] = "2020-01-24T10:00:00",
+    verbose: Optional[bool] = False,
+):
+    """
+    instance_id: Id of the air quality trained model.
+
+    secretfile: Path to the database secretfile.
+    """
+    logger = get_logger("Shortest path entrypoint")
+    if verbose:
+        logger.level = logging.DEBUG
+
+    # TODO change this to a AirQualityResultQuery
+    result_query = HexGridQuery(secretfile=secretfile)
+    logger.info("Querying results from an air quality model")
+    result_sql = result_query.query_results(
+        instance_id,
+        join_hexgrid=True,
+        output_type="sql",
+        start_time=start_time,
+        upto_time=upto_time,
+    )
+    logger.debug(result_sql)
+
+    gdf = gpd.GeoDataFrame.from_postgis(
+        result_sql, result_query.dbcnxn.engine, crs=4326
+    )
+    gdf = gdf.rename(columns=dict(geom="geometry"))
+    gdf.crs = "EPSG:4326"
+    # gdf = gpd.GeoDataFrame(result_df, crs=4326, geometry="geom")
+    logger.info("%s rows in hexgrid results", len(gdf))
+
+    if source is not None and target is not None:
+        # use bounding box of surrounding ellipse to limit graph size
+        box = ellipse_bounding_box((source[1], source[0]), (target[1], target[0]))
+        G: nx.MultiDiGraph = ox.graph_from_bbox(box[0], box[1], box[2], box[3])
+        G = G.to_directed()
```

Comment:
Remove for now. Create a new issue for implementing A*:

- need to change Euclidean distance for haversine.
- how would this work for pollution?
- how does this work for multi di graphs?